examples/maps_heatmaps/src/index.ts

```diff
@@ -21,36 +21,81 @@
 
 // Imports
 let map: google.maps.Map, heatmap: google.maps.visualization.HeatmapLayer;
-import * as firebase from 'firebase';
-import firebaseConfig from './firebase_config';
+import * as firebase from "firebase";
+import firebaseConfig from "./firebase_config";
+import * as queryDB from "./queryDB";
 
 // Creates the firebase app and gets a reference to firestore.
 console.log(firebaseConfig);
-var app = firebase.initializeApp(firebaseConfig);
-var database = app.firestore();
+const app = firebase.initializeApp(firebaseConfig);
+const database = app.firestore();
 
 function initMap(): void {
-
-
   map = new google.maps.Map(document.getElementById("map") as HTMLElement, {
     zoom: 13,
-    mapTypeId: "satellite"
+    mapTypeId: "satellite",
+    mapTypeControlOptions: {
+      style: google.maps.MapTypeControlStyle.HORIZONTAL_BAR,
+      position: google.maps.ControlPosition.TOP_CENTER,
+    },
   });
 
   // TODO shows how to connect to firestore and read data from there
-  let images = database.collection("test");
-  images.get().then(querySnapshot => {
-    if(!querySnapshot.empty) {
-      const image = querySnapshot.docs[0].data() as any;
-      map.setCenter(  { lat: image.coordinates.latitude, lng: image.coordinates.longitude });
+  const images = database.collection("test");
+  images.get().then((querySnapshot) => {
+    if (!querySnapshot.empty) {
+      const image = querySnapshot.docs[0].data();
+      map.setCenter({
+        lat: image.coordinates.latitude,
+        lng: image.coordinates.longitude,
+      });
     }
   });
 
-
   heatmap = new google.maps.visualization.HeatmapLayer({
-    data: getPoints(),
-    map: map
+    data: [],
+    map: map,
   });
+  map.addListener("center_changed", () => mapChanged());
+  map.addListener("zoom_changed", () => mapChanged());
+  function mapChanged() {
+    const center: google.maps.LatLng = map.getCenter();
+    const lat = center.lat();
+    const lng = center.lng();
+    const newCenter = new firebase.firestore.GeoPoint(lat, lng);
+    const quiredCollection = queryDB.getQuiredCollection(
+      newCenter,
+      getRadius(),
```

Comment:
Add:
`TODO: read the avilable labels from firestore`

Reply:
We added a TODO to the code.